### ISSUE TITLE
Is it possible to do PullToRefresh?

### ISSUE THREAD

**Issue by jun1st** [eBay/NMessenger#39]
Is it possible to do `pulltorefresh` ?

I may have a lot old messages and want be able to load it when pulling down.

**Comment by atainter** [comment]
Take a look at https://github.com/eBay/NMessenger#head-prefetching

**Comment by dodikk** [comment]
@jun1st , have you managed to implement your feature? If so, could you please share some tips and insights?

P.S. I'm stuck at not receiving any `batchFetchContent()` invocations even though I've overloaded the corresponding function and enabled the `self.messengerView.doesBatchFetch = true`
https://github.com/eBay/NMessenger/issues/97

**Comment by jun1st** [comment]
@dodikk Yes, I did get prefetch to work.

But I've move away from NMessager, so not able to find the code. 

If my memory works, the steps are set the property to true, hook the delegate, and implement the methods.

**Comment by kevinjonesevans** [comment]
Is it possible to get an example of using the head prefetching feature?? I have implemented the `batchFetchContent` method and it gets called, but I end up getting an error for not being on the main thread. After that I wrapped the body of my method in a `DispatchQueue.main.sync` block, which takes care of the first error. But then I run into the problem that my `messengerView.allMessages` returns nil.
